### PR TITLE
[FLAKE] Fixed UT pkg/kubelet/kuberuntime/logs/TestReadRotatedLog caused due to data race

### LINKED ISSUE #121564: Failing UT pkg/kubelet/kuberuntime/logs TestReadRotatedLog with DATARACE
### Which jobs are flaking?

Runs k8s with master golang:

[Failing-TestReadRotatedLog-DATARACE](https://prow.ppc64le-cloud.cis.ibm.net/view/gs/ppc64le-kubernetes/logs/postsubmit-master-golang-kubernetes-unit-test-ppc64le/1717239828946358272)

### Which tests are flaking?

test ```TestReadRotatedLog```

### Since when has it been flaking?

Failure was observed on Aug 25th, it is passing after that. However it can be reproduced with stress tool.

### Testgrid link

_No response_

### Reason for failure (if possible)

_No response_

### Anything else we need to know?

It is failing on both ppc and x86 architectures.

Steps to reproduce:

1. Setup up golang tip version.
2. Git clone kubernetes master

```
# stress ./logs.test -test.run ^TestReadRotatedLog

/tmp/go-stress-20231027T084859-422659708
==================
WARNING: DATA RACE
Write at 0x00c0000783b0 by goroutine 24:
  k8s.io/kubernetes/pkg/kubelet/kuberuntime/logs.TestReadRotatedLog()
      /root/kubernetes/pkg/kubelet/kuberuntime/logs/logs_test.go:280 +0x9b4
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1682 +0x18c
  testing.(*T).Run.gowrap1()
      /usr/local/go/src/testing/testing.go:1735 +0x64

Previous read at 0x00c0000783b0 by goroutine 25:
  k8s.io/kubernetes/pkg/kubelet/kuberuntime/logs.TestReadRotatedLog.func1()
      /root/kubernetes/pkg/kubelet/kuberuntime/logs/logs_test.go:243 +0xd0
  k8s.io/kubernetes/pkg/kubelet/kuberuntime/logs.TestReadRotatedLog.gowrap1()
      /root/kubernetes/pkg/kubelet/kuberuntime/logs/logs_test.go:244 +0x78

Goroutine 24 (running) created at:
  testing.(*T).Run()
      /usr/local/go/src/testing/testing.go:1735 +0x6ac
  testing.runTests.func1()
      /usr/local/go/src/testing/testing.go:2146 +0x90
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1682 +0x18c
  testing.runTests()
      /usr/local/go/src/testing/testing.go:2144 +0x7fc
  testing.(*M).Run()
      /usr/local/go/src/testing/testing.go:2012 +0xbe8
  main.main()
      _testmain.go:59 +0x2d0

Goroutine 25 (running) created at:
  k8s.io/kubernetes/pkg/kubelet/kuberuntime/logs.TestReadRotatedLog()
      /root/kubernetes/pkg/kubelet/kuberuntime/logs/logs_test.go:238 +0x5a4
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1682 +0x18c
  testing.(*T).Run.gowrap1()
      /usr/local/go/src/testing/testing.go:1735 +0x64
==================
--- FAIL: TestReadRotatedLog (0.10s)
    testing.go:1398: race detected during execution of test
FAIL


ERROR: exit status 1


```

### Relevant SIG(s)

/sig testing

### PR DESCRIPTION
**Pull request by srivastav-abhishek**
#### What type of PR is this?
/kind bug
/kind flake

#### What this PR does / why we need it:
This PR will fix data race issue observed for UT ```pkg/kubelet/kuberuntime/logs/TestReadRotatedLog```. Data race is happening as ```file``` is getting accessed without obtaining locks.

#### Which issue(s) this PR fixes:

Fixes #121564

#### Special notes for your reviewer:
**1. Race detector o/p before fix:**

```
[root@test kubernetes]# go test ./pkg/kubelet/kuberuntime/logs -c -race
[root@test kubernetes]# /usr/local/go/bin/stress ./logs.test -test.run ^TestReadRotatedLog

/tmp/go-stress-20231027T084859-422659708
==================
WARNING: DATA RACE
Write at 0x00c0000783b0 by goroutine 24:
  k8s.io/kubernetes/pkg/kubelet/kuberuntime/logs.TestReadRotatedLog()
      /root/kubernetes/pkg/kubelet/kuberuntime/logs/logs_test.go:280 +0x9b4
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1682 +0x18c
  testing.(*T).Run.gowrap1()
      /usr/local/go/src/testing/testing.go:1735 +0x64

Previous read at 0x00c0000783b0 by goroutine 25:
  k8s.io/kubernetes/pkg/kubelet/kuberuntime/logs.TestReadRotatedLog.func1()
      /root/kubernetes/pkg/kubelet/kuberuntime/logs/logs_test.go:243 +0xd0
  k8s.io/kubernetes/pkg/kubelet/kuberuntime/logs.TestReadRotatedLog.gowrap1()
      /root/kubernetes/pkg/kubelet/kuberuntime/logs/logs_test.go:244 +0x78

Goroutine 24 (running) created at:
  testing.(*T).Run()
      /usr/local/go/src/testing/testing.go:1735 +0x6ac
  testing.runTests.func1()
      /usr/local/go/src/testing/testing.go:2146 +0x90
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1682 +0x18c
  testing.runTests()
      /usr/local/go/src/testing/testing.go:2144 +0x7fc
  testing.(*M).Run()
      /usr/local/go/src/testing/testing.go:2012 +0xbe8
  main.main()
      _testmain.go:59 +0x2d0

Goroutine 25 (running) created at:
  k8s.io/kubernetes/pkg/kubelet/kuberuntime/logs.TestReadRotatedLog()
      /root/kubernetes/pkg/kubelet/kuberuntime/logs/logs_test.go:238 +0x5a4
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1682 +0x18c
  testing.(*T).Run.gowrap1()
      /usr/local/go/src/testing/testing.go:1735 +0x64
==================
--- FAIL: TestReadRotatedLog (0.10s)
    testing.go:1398: race detected during execution of test
FAIL
```

**2. Race detector o/p after the fix:**

```
[root@test kubernetes]# go test ./pkg/kubelet/kuberuntime/logs -c -race
[root@test kubernetes]# /usr/local/go/bin/stress ./logs.test -test.run ^TestReadRotatedLog
10s: 56 runs so far, 0 failures
15s: 88 runs so far, 0 failures
20s: 120 runs so far, 0 failures
25s: 152 runs so far, 0 failures
30s: 184 runs so far, 0 failures
35s: 218 runs so far, 0 failures
40s: 250 runs so far, 0 failures
.
.
.
.
.
.
23m30s: 9414 runs so far, 0 failures
23m35s: 9450 runs so far, 0 failures
23m40s: 9484 runs so far, 0 failures
23m45s: 9516 runs so far, 0 failures
23m50s: 9548 runs so far, 0 failures
23m55s: 9582 runs so far, 0 failures
24m0s: 9618 runs so far, 0 failures
24m5s: 9652 runs so far, 0 failures
```
#### Does this PR introduce a user-facing change?
NONE

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by aojea** [comment]
please paste the evidence in the description by running the `stress` tool with the binary compiled with the race detector

**Comment by srivastav-abhishek** [comment]
@aojea I have updated PR description with required details from ```stress``` tool.

File: pkg/kubelet/kuberuntime/logs/logs_test.go [hunk 1]
```diff
@@ -240,7 +241,10 @@ func TestReadRotatedLog(t *testing.T) {
 			Follow: true,
 		}
 		opts := NewLogOptions(&podLogOptions, time.Now())
-		ReadLogs(ctx, file.Name(), containerID, opts, fakeRuntimeService, stdoutBuf, stderrBuf)
+		mu.Lock()
```
Comment (aojea) [inline pkg/kubelet/kuberuntime/logs/logs_test.go:244]: you don't need a mutex, just capture the variable out of the gorutine is enough

Comment (srivastav-abhishek) [reply pkg/kubelet/kuberuntime/logs/logs_test.go:244]: Thanks for pointing it out. It is a better approach, I will do the necessary changes.

**Review by mkumatag — APPROVED** [review]
/lgtm

**Comment by srivastav-abhishek** [comment]
@aojea Please let us know if changes are fine.

**Pull request merged**